drivers/spi/spi_mcux_lpspi.c

```diff
@@ -585,14 +616,161 @@ static int spi_mcux_init(const struct device *dev)
 	return 0;
 }
 
+#ifdef CONFIG_SPI_RTIO
+static inline k_spinlock_key_t spi_spin_lock(const struct device *dev)
+{
+	struct spi_mcux_data *data = dev->data;
+
+	return k_spin_lock(&data->lock);
+}
+
+static inline void spi_spin_unlock(const struct device *dev, k_spinlock_key_t key)
+{
+	struct spi_mcux_data *data = dev->data;
+
+	k_spin_unlock(&data->lock, key);
+}
+
+
+static void spi_mcux_iodev_next(const struct device *dev, bool completion);
+
+static void spi_mcux_iodev_start(const struct device *dev)
+{
+	const struct spi_mcux_config *config = dev->config;
+	struct spi_mcux_data *data = dev->data;
+	struct rtio_sqe *sqe = &data->txn_curr->sqe;
+	struct spi_dt_spec *spi_dt_spec = sqe->iodev->data;
+	struct spi_config *spi_cfg = &spi_dt_spec->config;
+
+	LPSPI_Type *base = config->base;
+	lpspi_transfer_t transfer;
+	status_t status;
+
+	transfer.configFlags = kLPSPI_MasterPcsContinuous |
+			       (spi_cfg->slave << LPSPI_MASTER_PCS_SHIFT);
+
+	switch (sqe->op) {
+	case RTIO_OP_RX:
+		transfer.txData = NULL;
+		transfer.rxData = sqe->buf;
+		transfer.dataSize = sqe->buf_len;
+		break;
+	case RTIO_OP_TX:
+		transfer.rxData = NULL;
+		transfer.txData = sqe->buf;
+		transfer.dataSize = sqe->buf_len;
+		break;
+	case RTIO_OP_TINY_TX:
+		transfer.rxData = NULL;
+		transfer.txData = sqe->tiny_buf;
+		transfer.dataSize = sqe->tiny_buf_len;
+		break;
+	case RTIO_OP_TXRX:
+		transfer.txData = sqe->tx_buf;
+		transfer.rxData = sqe->rx_buf;
+		transfer.dataSize = sqe->txrx_buf_len;
+		break;
+	default:
+		LOG_ERR("Invalid op code %d for submission %p\n", sqe->op, (void *)sqe);
+		struct rtio_iodev_sqe *txn_head = data->txn_head;
+
+		spi_mcux_iodev_next(dev, true);
+		rtio_iodev_sqe_err(txn_head, -EINVAL);
+		spi_mcux_iodev_complete(dev, 0);
+		return;
+	}
+
+	data->transfer_len = transfer.dataSize;
+
+	k_spinlock_key_t key = spi_spin_lock(dev);
+
+	status = LPSPI_MasterTransferNonBlocking(base, &data->handle,
+						 &transfer);
+	spi_spin_unlock(dev, key);
+	if (status != kStatus_Success) {
+		LOG_ERR("Transfer could not start");
```

Comment:
rtio_iodev_sqe_err(txn_head, -EIO); or something likely needed here

Reply:
understood, added this call.